apps/web/src/api/hooks/notification-templates/useCreateOnboardingExperimentWorkflow.ts

```diff
@@ -0,0 +1,115 @@
+    onSuccess: (template) => {
+      navigate(parseUrl(ROUTES.WORKFLOWS_EDIT_TEMPLATEID, { templateId: template._id as string }));
+    },
+    onError: () => {
+      errorMessage('Failed to create onboarding experiment Workflow');
+    },
+  });
+
+  const { mutate: makePrimaryIntegration } = useMutation<IntegrationEntity, IResponseError, { id: string }>(
+    ({ id }) => setIntegrationAsPrimary(id),
+    {
+      onSuccess: () => {
+        queryClient.refetchQueries({
+          predicate: ({ queryKey }) =>
+            queryKey.includes(QueryKeys.integrationsList) || queryKey.includes(QueryKeys.activeIntegrations),
+        });
+      },
+      onError: () => {
+        errorMessage("Failed to update integration's primary status");
+      },
+    }
+  );
+
+  const { templates = [], loading: templatesLoading } = useTemplates(FIRST_100_WORKFLOWS);
+
+  const { integrations } = useIntegrations();
+
+  const onboardingExperimentWorkflow = 'Onboarding Workflow';
+
+  const createOnboardingExperimentWorkflow = useCallback(() => {
+    if (templatesLoading) return;
+
+    const onboardingExperimentWorkflowExists = templates.find((template) =>
+      template.name.includes(onboardingExperimentWorkflow)
+    );
+
+    const novuEmailIntegration = integrations?.find(
+      (integration) => integration.providerId === EmailProviderIdEnum.Novu
+    );
+
+    if (novuEmailIntegration && !novuEmailIntegration?.primary) {
+      makePrimaryIntegration({ id: novuEmailIntegration?._id as string });
+    }
+    // }
+    if (onboardingExperimentWorkflowExists) {
+      navigate(
+        parseUrl(ROUTES.WORKFLOWS_EDIT_TEMPLATEID, { templateId: onboardingExperimentWorkflowExists._id as string })
+      );
+    } else {
+      const payload = {
+        name: onboardingExperimentWorkflow,
+        notificationGroupId: groups[0]._id,
+        active: true,
+        draft: false,
+        critical: false,
+        tags: ['onboarding'],
+        steps: [
+          {
+            template: {
+              subject: 'Your first email notification from Novu!',
+              senderName: 'Novu Onboarding',
+              type: StepTypeEnum.EMAIL,
+              contentType: 'customHtml',
+              content:
+                // eslint-disable-next-line max-len
+                'It\'s that simple! <br/>Learn more about creating workflows <a href="https://docs.novu.co/workflows/notification-workflows">here</a>.',
+            },
+            uuid: uuid4(),
+            active: true,
+          },
+        ],
+      };
+
+      createNotificationTemplate({
+        template: payload as any,
+        params: { __source: 'Onboarding Experiment Workflow' },
+      });
+    }
+  }, [templatesLoading, templates, integrations, makePrimaryIntegration, navigate, groups, createNotificationTemplate]);
+
+  return {
+    createOnboardingExperimentWorkflow,
+    isLoading: isCreating,
+    isDisabled: areNotificationGroupLoading || isCreating,
```

Comment:
please update the states for these flags, when the templates and integrations are loading this flag should be also true, and the button should be disabled when it's loading